cdap-app-fabric/src/main/java/co/cask/cdap/lineage/field/LineageGraph.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.lineage.field;
+
+import co.cask.cdap.api.lineage.field.EndPoint;
+import co.cask.cdap.api.lineage.field.InputField;
+import co.cask.cdap.api.lineage.field.Operation;
+import co.cask.cdap.api.lineage.field.ReadOperation;
+import co.cask.cdap.api.lineage.field.TransformOperation;
+import co.cask.cdap.api.lineage.field.WriteOperation;
+import com.google.common.collect.Sets;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Represents field level lineage graph. Edges of the graph are simply collection of operations.
```

Comment:
How can an edge be a collection of operations?
>> Edge is just a single operation. Graph is a collection of edges or operations in our case, since edge represents operation.

Reply:
Then the Javadoc should say: An edge in the graph is simply an operation. But I am not really sure that this is true. Because if an operation reads two inputs, then it actually represents two edges (one from each of the input fields to the output field). I think that this API is not actually a graph, but a collection of operations, and it is less confusing if it is documented (and named) as such. 

The LineageGraph would be the other (field-oriented) representation that we may have to produce for the UI.

Reply:
Renamed the class to `FieldLineageInfo`.